google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/ui/BooleanTableModel.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.ui;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import javax.swing.event.TableModelEvent;
+import javax.swing.event.TableModelListener;
+import javax.swing.table.TableModel;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The {@link TableModel} for a 2 column headerless table where the first column is immutable and
+ * the second column in a boolean.
+ *
+ * @param <T> the type of the item in the first column.
+ */
+public class BooleanTableModel<T> implements TableModel {
+
+  private final SortedMap<T, Boolean> map;
+  private final List<TableModelListener> listeners = new ArrayList<>();
```

Comment:
This could be a bit easier to implement using `AbstractTableModel` - it contains support for listeners and set of `fire...` methods to fire table events.